contracts/index_executor/src/graph.rs

```diff
@@ -116,10 +116,13 @@ impl TryInto<index_graph::Graph> for Graph {
                             _ => return Err("Unsupported chain!"),
                         }
                     },
-                    native_asset: hex::decode(chain.native_asset.clone())
-                        .or(Err("InvalidInput"))?,
+                    native_asset: {
+                        let asset_id = chain.native_asset;
+                        let asset = &self.assets[asset_id as usize - 1];
```

Comment:
Yes, long story, it's about why we need two graph here. To make it short, we do this is for adapt the off-chain database index. @tenheadedlion can add more details

Reply:
I think @h4x3rotab was not mentioning the rationale of the 2 graphs(for that you can refer to https://github.com/Phala-Network/index-contract/pull/23). here's the reason why we need to shift by 1:

the graph is generated off-chain by a tool "tablizer" which uses Database to manage entities, by default entity ids start from 1. my first solution is to reserve an empty slot at position 0, which saves us the effort of shifting by 1, however, @tolak said that would cause much trouble when converting one graph to another. in my opinion, that affects readability, but it works, so we end up like this(previous discussion: https://github.com/Phala-Network/index-contract/pull/38)